### PR TITLE
Fixed tabs from switching windows

### PR DESCRIPTION
**Pull request by KaifHalak**
added 
`const currentWindowObject = await chrome.windows.getCurrent() `

and updated the IF condition to:
` if (existingTab.groupId !== chrome.tabs.TAB_ID_NONE && existingTab.windowId === currentWindowObject.id)`

This will ensure that the tab and its associated group are on the same window

### CODE REVIEW

**Comment by AdamKmet1997** [comment]
I'm still experiencing the same bug as before with these code changes.

Also what it does is it groups all tabs on Window A and the single tabs are pushed to Window B.

I believe that your changes are not applied to just one window at a time but more globally to all windows thats why they are still looked as one window and tabs are moving inbetween windows.

**Comment by AdamKmet1997** [comment]
Could you also In the description attach a video of your implemented changes? Makes it easier to review the changes, because I can compare what you see with what I see

**Comment by KaifHalak** [comment]
Regarding your first comment, can you please share a video of the issue because on my end it works (I use Google Chrome).

As for the second comment, sure I will try to do that when I make large changes.

**Comment by AdamKmet1997** [comment]
> Regarding your first comment, can you please share a video of the issue because on my end it works (I use Google Chrome).

Sure have a look at the behaviour

Youtube is moved to Window 1 because the group already exists there 

[screen-capture.webm](https://github.com/AdamKmet1997/Group-My-Tabs/assets/32263448/cf1f099b-022b-4760-89a2-23efce2f1fb9)


I think youre close to solving it BUT you might also let the helper functions like `queryGroups`, `queryAllTabs` know that they need to also look at windowid?

**Comment by KaifHalak** [comment]
Ohh Ok Ok I get it. I will get right onto it

**Comment by KaifHalak** [comment]
Updated the code. Please check my second commit (2.0)

I added the following function to retrieve the current active window.


Made the following changes:

lastFocusedWindowObject.tabs contains a list of all tabs in that window, making tabs.query redundant.

Added window id parameter when fetching groups and its tabs


Changed variable name and once again used lastFocusedWindowObject.tabs instead of tabs.query


Here's a vid of the extension in action

[Fixed.webm](https://github.com/AdamKmet1997/Group-My-Tabs/assets/131174445/0fde4c97-86a1-45a4-adef-33f49299355c)

File: background.js [hunk 1]
```diff
@@ -10,37 +10,56 @@ function getGroupForTab(groups, tab) {
...
   };
 
+  // Get all tabs in the CURRENT Window
   const queryAllTabs = () => {
-    return new Promise((resolve) => {
-      chrome.tabs.query({}, (tabs) => {
-        resolve(tabs);
-      });
-    });
+    // return new Promise((resolve) => {
```
Comment (AdamKmet1997) [inline background.js:53]: You can delete all this commented out code if no longer needed

**Review by AdamKmet1997 — APPROVED** [review]
After my testing I can say that it matches the video that you have attached and everything works as desired.
There are few things like Leaving commented out code  in the PR for no reason, if its commented out it can be deleted.

But i wont let this delay the merge and I'll clean it up later on 

Great job on your first Open source PR this will go right into the next release

hopefully many more to come

File: background.js (hunk 1 above)
Comment (KaifHalak) [reply background.js:53]: Will keep that in mind for the next time👍

**Pull request merged**